pkg/deploy/registry/deployconfig/strategy.go

```diff
@@ -105,12 +114,44 @@ func (strategy) CheckGracefulDelete(obj runtime.Object, options *metav1.DeleteOp
 	return false
 }
 
+// legacyStrategy implements behavior for DeploymentConfig objects in the legacy API
+type legacyStrategy struct {
+	strategy
+}
+
+// PrepareForCreate delegates to the common strategy.
+func (s legacyStrategy) PrepareForCreate(ctx apirequest.Context, obj runtime.Object) {
+	s.strategy.PrepareForCreate(ctx, obj)
+}
+
+// DefaultGarbageCollectionPolicy for legacy DeploymentConfigs will orphan dependents.
+func (s legacyStrategy) DefaultGarbageCollectionPolicy() rest.GarbageCollectionPolicy {
+	return rest.OrphanDependents
+}
+
+// groupStrategy implements behavior for DeploymentConfig objects in the Group API
+type groupStrategy struct {
+	strategy
+}
+
+// PrepareForCreate delegates to the common strategy and sets defaults applicable only to Group API
+func (s groupStrategy) PrepareForCreate(ctx apirequest.Context, obj runtime.Object) {
+	s.strategy.PrepareForCreate(ctx, obj)
+
+	dc := obj.(*deployapi.DeploymentConfig)
+
+	if dc.Spec.RevisionHistoryLimit == nil {
```

Comment:
Split this into its own method for sanity.  `func AppsV1DeploymentConfigLayeredDefaults(*deployapi.DeploymentConfig)` and add a good godoc comment.